pkg/cloud/services/network/subnets.go

```diff
@@ -248,13 +248,13 @@ func (s *Service) deleteSubnets() error {
 	}
 
 	// Describe subnets in the vpc.
-	existing, err := s.describeVpcSubnets()
+	existing, err := s.describeSubnets()
 	if err != nil {
 		return err
 	}
 
-	for _, sn := range existing {
-		if err := s.deleteSubnet(sn.ID); err != nil {
+	for _, sn := range existing.Subnets {
+		if err := s.deleteSubnet(*sn.SubnetId); err != nil {
```

Comment:
```suggestion
		if err := s.deleteSubnet(aws.StringValue(sn.SubnetId)); err != nil {
```

Reply:
Thanks, didn't know about this function. Updated the PR with the suggestion.